tests/services/analysis_starter/test_balsamic_configurator.py

```diff
@@ -0,0 +1,439 @@
+from pathlib import Path
+from unittest.mock import Mock, create_autospec
+
+import pytest
+
+from cg.constants import SexOptions
+from cg.constants.priority import SlurmQos
+from cg.exc import (
+    BalsamicInconsistentSexError,
+    BalsamicMissingTumorError,
+    BedFileNotFoundError,
+    CaseNotConfiguredError,
+)
+from cg.services.analysis_starter.configurator.implementations.balsamic import BalsamicConfigurator
+from cg.services.analysis_starter.configurator.models.balsamic import (
+    BalsamicCaseConfig,
+    BalsamicConfigInput,
+)
+from cg.store.models import Application, ApplicationVersion, BedVersion, Case, Sample
+
+PANEL_ONLY_FIELDS = ["soft_filter_normal", "panel_bed", "pon_cnn", "exome"]
+WGS_ONLY_FIELDS = ["genome_interval", "gens_coverage_pon"]
+
+
+def test__get_pon(balsamic_configurator, tmp_path):
+    # GIVEN a balsamic configurator
+    balsamic_configurator.pon_directory = tmp_path
+    # GIVEN a bed file
+    panel_name = "GMS_duck"
+    bed_file = Path(balsamic_configurator.bed_directory, f"{panel_name}.bed")
+
+    # GIVEN two matching PON files
+    old_pon_path = Path(tmp_path, f"{panel_name}_CNVkit_PON_reference_v1.cnn")
+    new_pon_path = Path(tmp_path, f"{panel_name}_CNVkit_PON_reference_v2.cnn")
+    for pon_path in [old_pon_path, new_pon_path]:
+        pon_path.touch()
+
+    # WHEN getting the pon path
+    pon_path = balsamic_configurator._get_pon(bed_file)
+
+    # THEN the returned path should be the latest version
+    assert pon_path == new_pon_path
+
+
+def test__get_pon_no_match(balsamic_configurator: BalsamicConfigurator, tmp_path: Path):
+    # GIVEN a balsamic configurator
+    balsamic_configurator.pon_directory = tmp_path
+
+    # GIVEN a bed file with no matching PON files
+    panel_name = "GMS_duck"
+    bed_file = Path(balsamic_configurator.bed_directory, f"{panel_name}.bed")
+
+    # WHEN getting the pon path
+    # THEN it should raise a FileNotFoundError
+    with pytest.raises(FileNotFoundError):
+        balsamic_configurator._get_pon(bed_file)
+
+
+def test__get_patient_sex():
+    """Tests that the correct sex is returned when all samples have the same sex"""
+    # GIVEN a case with two samples
+    case_with_male = create_autospec(Case)
+
+    # GIVEN that the samples have the same sex
+    sample1 = create_autospec(Sample, sex=SexOptions.MALE)
+    sample2 = create_autospec(Sample, sex=SexOptions.MALE)
+    case_with_male.samples = [sample1, sample2]
```

Comment:
I use this pattern in many of the tests. Does it make the test clearer or should I hide it away in some fixture?